### PR TITLE
neon/sqadd: initial implementation

### PR DESCRIPTION
**Pull request by Glitch18**
Implements neon/sqadd

### CODE REVIEW

File: simde/arm/neon/sqadd.h [hunk 1]
```diff
@@ -0,0 +1,301 @@
...
+ * EXPRESS OR IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF
+ * MERCHANTABILITY, FITNESS FOR A PARTICULAR PURPOSE AND
+ * NONINFRINGEMENT. IN NO EVENT SHALL THE AUTHORS OR COPYRIGHT HOLDERS
+ * BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER LIABILITY, WHETHER IN AN
+ * ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM, OUT OF OR IN
+ * CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN THE
+ * SOFTWARE.
+ *
+ * Copyright:
+ *   2021      Evan Nemerson <evan@nemerson.com>
```
Comment (mr-c) [inline simde/arm/neon/sqadd.h:24]:
```suggestion
```
🙂

Comment (Glitch18) [reply simde/arm/neon/sqadd.h]: Done

**Comment by nemequ** [comment]
This looks great, thanks!

**Comment by nemequ** [comment]
Ah, it looks like [Drone turned up a problem](https://cloud.drone.io/simd-everywhere/simde/1881/2/1): clang 6.0 returned a signed result instead of an unsigned result for the scalar functions.  The fix is easy; just cast the return value to the correct type.  I'll take care of [the patch](https://github.com/simd-everywhere/simde/commit/332a80feaf6efd12f122cd0dec88dd82960f7ec8) since it's a bit of a pain to test.

Unfortunately we often have to work around compiler bugs in SIMDe ☹.

**Pull request closed without merging**